rotkehlchen/chain/evm/decoding/extrafi/decoder.py

```diff
@@ -0,0 +1,350 @@
+                event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                event.counterparty = CPT_EXTRAFI
+                event.notes = f'Lock {amount} EXTRA until {timestamp_to_date(locktime, formatstr="%d/%m/%Y %H:%M:%S")}'  # noqa: E501
+                event.product = EvmProduct.STAKING
+                break
+        else:
+            log.error(f'Failed to find lock of EXTRA in {context.transaction}')
+
+        return DEFAULT_DECODING_OUTPUT
+
+    def _partially_close_farm_position(self, context: DecoderContext) -> DecodingOutput:
+        """Close a position in a farm partially"""
+        manager = hex_or_bytes_to_address(context.tx_log.topics[3])
+        amount_0_received = hex_or_bytes_to_int(context.tx_log.data[32:64])
+        amount_1_received = hex_or_bytes_to_int(context.tx_log.data[64:96])
+
+        for event in context.decoded_events:
+            if event.location_label != manager:
+                continue
+
+            if (
+                event.maybe_get_direction() == EventDirection.IN and
+                (
+                    asset_normalized_value(amount_0_received, event.asset) == event.balance.amount or  # type: ignore  # noqa: E501
+                    asset_normalized_value(amount_1_received, event.asset) == event.balance.amount  # type: ignore
+                )
+            ):
+                event.counterparty = CPT_EXTRAFI
+                event.event_type = HistoryEventType.WITHDRAWAL
+                event.event_subtype = HistoryEventSubType.REMOVE_ASSET
+                event.notes = f"Withdraw {event.balance.amount} {event.asset.symbol_or_name()} from Extrafi's farm"  # noqa: E501
+                break
+        else:
+            log.error(f'Could not find withdrawal event for extrafi at {context.transaction}')
+
+        return DEFAULT_DECODING_OUTPUT
+
+    def _invest_into_farm_position(self, context: DecoderContext) -> DecodingOutput:
+        manager = hex_or_bytes_to_address(context.tx_log.topics[3])
+        amount_0_invested = hex_or_bytes_to_int(context.tx_log.data[0:32])
+        amount_1_invested = hex_or_bytes_to_int(context.tx_log.data[32:64])
+
+        for event in context.decoded_events:
+            if (
+                event.maybe_get_direction() == EventDirection.OUT and
+                event.location_label == manager and
+                (
+                    event.balance.amount == asset_normalized_value(amount_0_invested, event.asset) or  # type: ignore  # noqa: E501
+                    event.balance.amount == asset_normalized_value(amount_1_invested, event.asset)  # type: ignore
+                )
+            ):
+                event.counterparty = CPT_EXTRAFI
+                event.event_type = HistoryEventType.DEPOSIT
+                event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                event.notes = f'Deposit {event.balance.amount} {event.asset.symbol_or_name()} in Extrafi'  # noqa: E501
```

Comment:
same about farm name or "a farm" here. Otherwise user has no ideaif this a pool or those farms you mention